### PR TITLE
Add separate generation of CSLC-S1 static layers

### PR DESCRIPTION
**Pull request by vbrancat**
This PR allows the SAS CSLC-S1 to independently generate the static layers product. Therefore, the new CSLC-S1 SAS behavior is the following:

1. when `product_type: CSLC_S1`, the SAS generates only the CSLC-S1 product
2. when `product_type: CSLC_S1_STATIC`, the SAS generates only the CSLC-S1 static layers product

Flags to run rdr2geo and geocode the static layers have been removed from the schema. The default SAS `product_type` has been set to always generate the CSLC-S1 product.

Note: would be good if this PR can be merged ASAP so that the integration of the LOS vectors into the CSLC-S1 static layers workflow will be much easier. Thank you :)

### CODE REVIEW

**Comment by seongsujeong** [comment]
A comment not much related to the point of this PR:
I see the checks in the recent PRs in this repo are failing on "Read the Docs build". Will merging the PR #170 solve the issue?

File: src/compass/s1_static_layers.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from compass.utils.helpers import bursts_grouping_generator, get_module_name
+from compass import s1_rdr2geo
+from compass import s1_geocode_metadata
+
+
+def _make_rdr2geo_cfg(yaml_runconfig_str):
+    '''
+    Make a rdr2geo specific runconfig with latitude, longitude, and height
+    layers enabled for static layer product generation while preserving all
+    other rdr2geo config settings
```
Comment (seongsujeong) [inline src/compass/s1_static_layers.py:17]: I think adding the description about the input / output will be helpful.

File: src/compass/s1_static_layers.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    # Load a GeoRunConfig from modified yaml cfg string
+    rdr2geo_cfg = GeoRunConfig.load_from_yaml(yaml_runconfig_str,
+                                              workflow_name='s1_cslc_geo')
+
+    return rdr2geo_cfg
+
+
+def run(cfg: GeoRunConfig):
+    '''
+    Run geocode burst workflow with user-defined
```
Comment (seongsujeong) [inline src/compass/s1_static_layers.py:33]: Please modify the description accordingly for the static layer workflow.

**Review by seongsujeong — COMMENTED** [review]
The issues I've pointed here are minor issues. My concern about this PR is the conflict between the array mode PR in #127. The array mode PR makes significant changes in `s1_geocode_slc.py`, which this PR does as well. It's great if you have a good solution for that, but wanted to give some heads up about what I foresee.

**Review by LiangJYu — APPROVED** [review]
LGTM aside from comments by @seongsujeong

**Comment by vbrancat** [comment]
> 

This is a question for @scottstanie!

File: src/compass/schemas/s1_cslc_geo.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ runconfig:
            product_specification_version: str(required=False)
 
         primary_executable:
-          product_type: enum('CSLC_S1')
+          product_type: enum('CSLC_S1', 'CSLC_S1_SL')
```
[the remark this answers is not in the document]
Comment (vbrancat) [reply src/compass/schemas/s1_cslc_geo.yaml:43]: Good point. I remember there were discussion on this. I will implement this change as this is likely to be happening.

**Comment by vbrancat** [comment]
@LiangJYu and @seongsujeong all your comments should have been addressed. Please, have another look to this PR. One thing I have notice is that there is something off with measuring the runtime of the static layers. This is what I get by running the static layers workflow:

```
journal: elapsed time (GEO-IN) [s]: 6.704
journal: s1_geocode_metadata burst successfully ran in -19067 days, 8:07:32 (hr:min:sec)
journal: s1_static_layers burst successfully ran in 0:01:30 (hr:min:sec)
```
Not sure why `s1_geocode_metadata` gives that awful result :s.

**Comment by LiangJYu** [comment]
> @LiangJYu and @seongsujeong all your comments should have been addressed. Please, have another look to this PR. One thing I have notice is that there is something off with measuring the runtime of the static layers. This is what I get by running the static layers workflow:
> 
> ```
> journal: elapsed time (GEO-IN) [s]: 6.704
> journal: s1_geocode_metadata burst successfully ran in -19067 days, 8:07:32 (hr:min:sec)
> journal: s1_static_layers burst successfully ran in 0:01:30 (hr:min:sec)
> ```
> 
> Not sure why `s1_geocode_metadata` gives that awful result :s.

I'll look into it

File: src/compass/s1_static_layers.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    cfg: GeoRunConfig
+        GeoRunConfig object with user runconfig options
+    '''
+
+    module_name = get_module_name(__file__)
+    info_channel = journal.info(f"{module_name}.run")
+    info_channel.log(f"Starting {module_name} burst")
+
+    # Start tracking processing time
+    t_start = time.time()
```
Comment (LiangJYu) [inline src/compass/s1_static_layers.py:59]:
```suggestion
    t_start = time.perf_counter()
```
For whatever reason the change above also needs to be done [here](https://github.com/opera-adt/COMPASS/blob/de33cfad9692efc08e9e53296a2e6d1fdad410df/src/compass/s1_geocode_metadata.py#L84) to address [this](https://github.com/opera-adt/COMPASS/pull/177#issuecomment-1638703373).

**Comment by vbrancat** [comment]
@seongsujeong do you mind having another look to this PR? Thanks :)

**Review by seongsujeong — APPROVED** [review]
LGTM 👍

**Pull request merged**